provider/pkg/gen/params.go

```diff
@@ -21,7 +22,10 @@ func FlattenParams(
 ) (map[string]interface{}, error) {
 	out := map[string]interface{}{}
 	converter := resources.SdkShapeConverter{Types: types}
-	for k, v := range input {
+	// Sort the keys to ensure consistent ordering
+	sortedKeys := codegen.SortedKeys(input)
+	for _, k := range sortedKeys {
```

Comment:
We may as well inline this.
```suggestion
	for _, k := range codegen.SortedKeys(input) {
```

Reply:
Done 👍